app/src/lib/stores/app-store.ts

```diff
@@ -2206,6 +2212,14 @@ export class AppStore extends TypedBaseStore<IAppState> {
         PullRequestSuggestedNextAction
       ) ?? defaultPullRequestSuggestedNextAction
 
+    // Current users will have underlines disabled by default. In the future
+    // this will change to be true by default.
+    if (getBoolean(underlineLinksKey) === undefined) {
```

Comment:
```suggestion
   const currentVersion = new semver.SemVer(getVersion())
   // Assumes this will be released with 3.3.9 and so this will lead to 2 versions in prod before it becomes default new behavior for new users.
   const hasUnderlinesBeenReleasedForAFewVersions =  semver.gt(new semver.SemVer(`3.3.11`), currentVersion)

    if (!hasUnderlinesBeenReleasedForAFewVersions && getBoolean(underlineLinksKey) === undefined && __RELEASE_CHANNEL__ === 'production ) {
```
WARNING ^ Suggestion will break things (need imports and linter run).. just put it in a suggestion to better communicate it.

Realized that we will want this to go straight to in effect for beta and test users for accessibility testing. I think it is reasonable for beta users to not get opted out of this feature so it can be tested before it hits production. 

Additionally, this only becomes default behavior for Accessibility if we remember to take this check out in a future release. I propose we add a check in that says that once we get past version 3.3.11 (2 versions), we go ahead and ignore this check.

Reply:
Other thoughts... maybe this whole check should be in a function with documentation. :P

Reply:
@tidy-dev Why the `__RELEASE_CHANNEL__ === 'production'`? I've been working on this change and don't fully understand that suggestion.

Reply:
Premise: I think we decided not to force adoption to existing users since underlines are only required for accessibility in the couple places we have secondary-text - majority of our links this is not necessary for accessibility.  I am suggesting that we hold to that for production, but not for beta so that our beta users do get forced to see the underlines because we want any feature to be tested as much as possible before production and it makes it easier for audit verification if it is on by default for our non-production builds. 

Logic explanation:
If we released this in our next beta and production. Via line 2221 and if this check wasn't here, any user to adopt that version will see the underlines.  To prevent forced adoption, we add this "if no local storage, set to false" check in. We ship this for a few versions to get our current users who adopt it set to false so that when we ship without this check, new users (those who have no local storage) will now have underlines by default, but our existing users will have a local storage setting for not having underlines. (Technically there is some overlap in those time frames of new users/existing users.. but this covers the majority).

My suggestion here, just makes it so that beta and test users are forced to adopt underlines, as the setting to false will only apply to production users. The version check bit just makes it so we forget to remove this check, in a few prod versions, new users will see the underline by default.. which is what is required to fully meet accessibility requirements. 


Other thoughts if you would like to do away with this entirely..... _sorry for the back and forth_ ....  The checks work I am doing we cannot do the above because not having a check is required by default all the time where for the links it is only required in a couple of small use cases. I was talking with Sergio last week and I think it makes sense to have the checks on/off setting in your accessibility settings tab since it is likely a one time setting.. not something to toggle between frequently... like something you would put more prominently say in the diff settings or context menu. (tho I am still a little turn on having a quick way) That said, I also think with the checks work we need a pretty prominent announcement about the checks are able to be turned off by those who don't like it. We could plan to ship your underline and this checks in the same production release so that the underline setting also gets major announcement. That way we do not have do this rigamarole of settings to false by default and just let users choose to toggle the settings if they do not like them. Probably worth discussing in the team sync.

Reply:
@tidy-dev Circling back to this, does the last comment mean that maybe it's an OK idea to just ship it as on by default for users who have no preference saved?

Reply:
Yes. I believe that is where the convo this has landed. We just need to make sure it is feature flagged to beta so that we hold back shipping to production until that announcement is prepped.